src/peer-store/book.js

```diff
@@ -10,16 +10,19 @@ const {
 const passthrough = data => data
 
 /**
- * The Book is the skeleton for the PeerStore books.
+ * @typedef {import('./')} PeerStore
  */
+
 class Book {
   /**
+   * The Book is the skeleton for the PeerStore books.
```

Comment:
Yes, it should probably have a composed template. I started doing it and it was working good here. However, when I tried to integrate this in lodestar, I was getting issues on building their project. You have more information on this in https://github.com/libp2p/js-libp2p/pull/802#issuecomment-738128860 . I have it done in #821 and I already talked with @hugomrdias about it. It seems like a typescript issue (I need to open an issue there), and we should track it before changing it here.

Reply:
I have looked into this & turns out this was different issue fixed in typescript@4.1.x. So, I have submitted #831 against #821, which addresses this problem and fixes few type miss matches that have came up.

I think it would be good to take a look at that pull as it also uncovered some issues that I have missed when reviewing this.